### PR TITLE
Update and add documentation files

### PR DESCRIPTION
**Pull request by ohltyler**
Signed-off-by: Tyler Ohlsen <ohltyler@amazon.com>

### Description

This PR standardizes some existing docs, and adds some new ones, to meet the repository standards listed [here](https://github.com/opensearch-project/.github/issues/8).

Changes include:
- Linking to standardized `CONTIBUTING` doc
- Refactoring env setup / build / test etc. into a DEVELOPER_GUIDE doc
- Standardizing MAINTAINERS
- Adding OpenSearch logo + cleaning up README
- Add standardized RELEASING
- Add standardized SECURITY

Note: ADMINS file is not added yet - this will be done by the admins themselves.

### Check List

- [x] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,105 +1,4 @@
...
-If you discover a potential security issue in this project we ask that you notify AWS/Amazon Security via our [vulnerability reporting page](http://aws.amazon.com/security/vulnerability-reporting/). Please do **not** create a public GitHub issue.
-
-
-## Licensing
-
-See the [LICENSE](./LICENSE.txt) file for our project's licensing. We will ask you to confirm the licensing of your contribution.
+## Contributing to this Project
 
+OpenSearch is a community project that is built and maintained by people just like **you**.
+[This document](https://github.com/opensearch-project/.github/blob/main/CONTRIBUTING.md) explains how you can contribute to this and related projects.
```
Comment (ylwu-amzn) [inline CONTRIBUTING.md:4]: I remember we have similar discussion before about whether maintain a hard copy of such common shared doc. Do we finalize the decision to just share a link in each plugin?

Comment (ohltyler) [reply CONTRIBUTING.md:4]: Yes - see [here](https://github.com/opensearch-project/.github/issues/8)

File: MAINTAINERS.md [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
+| Hanguang Zhang          | [zhanghg08](https://github.com/zhanghg08)               | Amazon      |
+| Jack Mazanec            | [jmazanec15](https://github.com/jmazanec15)             | Amazon      |
+| Jing Zhang              | [jngz-es](https://github.com/jngz-es)                   | Amazon      |
+| Kaituo Li               | [kaituo](https://github.com/kaituo)                     | Amazon      |
+| Lai Jiang               | [wnbts](https://github.com/wnbts)                       | Amazon      |
+| Sarat Vemulapalli       | [saratvemulapalli](https://github.com/saratvemulapalli) | Amazon      |
+| Tyler Ohlsen            | [ohtyler](https://github.com/ohltyler)                  | Amazon      |
+| Vamshi Vijay Nakkirtha  | [vamshin](https://github.com/vamshin)                   | Amazon      |
+| Vijayan Balasubramanian | [VijayanB](https://github.com/VijayanB)                 | Amazon      |
+| Weicong Sun             | [weicongs-amazon](https://github.com/weicongs-amazon)   | Amazon      |
```
Comment (kaituo) [inline MAINTAINERS.md:15]: Weicong already left.  Do we want to remove him?

**Review by kaituo — APPROVED** [review]

File: MAINTAINERS.md (hunk 2 above)
Comment (ohltyler) [reply MAINTAINERS.md]: good point - updated

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks for the change

**Pull request merged**